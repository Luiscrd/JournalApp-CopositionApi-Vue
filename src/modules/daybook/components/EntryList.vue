<template>
    <div class="entry-list-container">
        <div class="box-input px-2 pt-2 pb-2">
            <input class="form-control" type="text" placeholder="Buscar entradas">
        </div>
        <div class="entry-scrollarea">
            <Entry v-for="entry in 100" :key="entry" />
        </div>
    </div>
</template>

<script>
import { defineAsyncComponent } from 'vue'

export default {
    components: {
        Entry: defineAsyncComponent(() => import('@/modules/daybook/components/Entry.vue')),
    }
}
</script>

<style lang="scss" scoped>
.entry-list-container {
    position: relative;
    border-right: 1px solid #62B8FC;
    height: calc(100vh - 217px);
}

.box-input{
    border-bottom: 1px solid #62B8FC;
}

.entry-scrollarea {
    height: calc(100vh - 271px);
    overflow-y: scroll;

    &::-webkit-scrollbar {
        width: 8px;
        height: 17px;
        display: none;
    }

    &::-webkit-scrollbar-thumb {
        background: #ccc;
        border-radius: 4px;
    }

    &::-webkit-scrollbar-thumb:active {
        background-color: #999999;
    }

    &::-webkit-scrollbar-track {
        background: #e1e1e1;
        border-radius: 4px;
    }
}
</style>