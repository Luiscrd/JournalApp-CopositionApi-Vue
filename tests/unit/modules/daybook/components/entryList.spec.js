import { shallowMount } from '@vue/test-utils'
import EntryList from '@/modules/daybook/components/EntryList.vue'
import { createStore } from 'vuex'
import { journalState } from '../../../mook-data/test-journal-state'
import journal from '@/modules/daybook/store/journal/index'


describe('Pruebas en el Fab component', () => {

    const createVuexStore = (initialState) => createStore({
        modules: {
            journal: {
                ...journal,
                state: { ...initialState }
            }
        }
    })

    const store = createVuexStore(journalState)

    const mockRouter = {
        push: jest.fn(),
    }

    let wrapper

    beforeEach(() => {

        jest.clearAllMocks()

        wrapper = shallowMount(EntryList, {
            global: {
                mocks: {
                   $router: mockRouter
                },
                plugins: [store]
            }
        })

    })

    test('Debe renderizar el componente correctamente', () => {

        expect(wrapper.html()).toMatchSnapshot()

    });

    test('Debe de llamar el getEntriesByTerm sin termino y mostrar 2 entradas', () => {

       expect(wrapper.findAll('entry-stub').length).toBe(2)

    });

    test('Debe de llamar el getEntriesByTerm y filtrtar las entradas', async () => {

        const input = wrapper.find('input')

        await input.setValue('abcdef')

        expect(wrapper.findAll('entry-stub').length).toBe(1)
 
     });

     test('El botÃ³n nuevo debe redireccionar a /new', () => {

        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenCalled()
 
     });


})