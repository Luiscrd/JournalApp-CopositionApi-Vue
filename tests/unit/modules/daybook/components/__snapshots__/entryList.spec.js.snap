// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Fab component Debe renderizar el componente correctamente 1`] = `
<div class="entry-list-container">
  <div class="box-input px-2 pt-2 pb-2"><input class="form-control" type="text" placeholder="Buscar entradas"></div>
  <div class="box-button d-flex flex-column p-2"><button type="button" class="btn btn-primary">Nueva Entrada<i class="fa fa-plus-circle ms-3"></i></button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
